### PR TITLE
Add Visual Studio projects, scripts, and docs

### PR DESCRIPTION
**Pull request by riverar**

### CODE REVIEW

**Comment by riverar** [comment]
Am refreshing the docs, will be done with this tonight.

File: lib/lib.vcxproj [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    <CustomBuild Include="libdtrace\common\dt_grammar.y">
+      <Command>"$(SolutionDir)\releng\external\tools\bin\win_bison.exe" -y "-o$(ProjectDir)\libdtrace\compat\win32\prebuilt\%(FileName).c" -d -y "%(FullPath)"</Command>
+      <Message>Bison: %(Identity)</Message>
+      <Outputs>$(ProjectDir)\libdtrace\compat\win32\prebuilt\%(Filename).c;$(ProjectDir)\libdtrace\compat\win32\prebuilt\%(Filename).h</Outputs>
+      <OutputItemType></OutputItemType>
+    </CustomBuild>
+    <CustomBuild Include="libdtrace\common\dt_lex.l">
+      <Command>"$(SolutionDir)releng\external\tools\bin\win_flex.exe" -l --wincompat "-o$(ProjectDir)libdtrace\compat\win32\prebuilt\%(FileName).c" "%(FullPath)"</Command>
+      <Message>Flex: %(Identity)</Message>
+      <Outputs>$(ProjectDir)libdtrace\compat\win32\prebuilt\%(Filename).c</Outputs>
```
Comment (ashedel) [inline lib/lib.vcxproj]: Would be great to use $(IntDir) here and in similar places instead to keep all autogenerated files in the same location.

Comment (riverar) [reply lib/lib.vcxproj]: I think I went down that road initially, then ran into conflicts as dtrace.obj can ambiguously refer to dtrace.exe or dtrace.dll

Comment (riverar) [reply lib/lib.vcxproj]: @ashedel (Oops, hit the wrong button.) Sounds good, will do now.

**Comment by gopimsft** [comment]
Validated the build was successful and the newly generated binaries works on 19H1 OS.

**Comment by gopimsft** [comment]
Validated the build was successful and the newly generated binaries works in 19H1 OS.

**Pull request merged**